app/roles/api/init.lua

```diff
@@ -0,0 +1,212 @@
+    end
+
+    if latitude < -90 or latitude > 90 then
+        return nil, err_request_params:new('latitude have wrong value: ' .. latitude)
+    end
+
+    return true
+end
+
+-- делаем запрос в тарантул и пытаемся получить данные
+-- если данные есть:
+--      отдаем их клиенту
+-- если данных нет:
+--      делаем запрос к сервису прогноза погоды
+--      сохраняем в тарантул
+--      отдаем клиенту
+local function http_weather_get_handler(req)
+    local valid, err = validate_request_params(req)
+    if not valid then
+        log.info(err)
+
+        return bad_response(req, err.err, 422)
+    end
+
+    local latitude = req:query_param('latitude')
+    local longitude = req:query_param('longitude')
+
+    longitude = tonumber(longitude)
+    latitude = tonumber(latitude)
+
+    -- get router and create bucket id
+    local router = cartridge.service_get('vshard-router').get()
+    local bucket_by_data = calc_bucket_by_coords(longitude, latitude)
+    local bucket_id = router:bucket_id(bucket_by_data)
+
+    -- trying to get weather forecast from cache (taranrool)
+    local t_resp, err = err_vshard_router:pcall(
+        router.call,
+        router,
+        bucket_id,
+        'read',
+        'weather_forecast_get',
+        {longitude, latitude}
+    )
+    -- tarantool request error
+    if err then
+        log.error(err)
+
+        return bad_response(req, err.err, 500)
+    end
+
+    -- successfully retrieved data from cache (taranrool)
+    if t_resp then
+        return ok_response(req, json.decode(t_resp.raw_data))
+    end
+
+    -- make request to weather service
+    local raw_forecast, err = err_http_client:pcall(
+        make_weather_query,
+        longitude,
+        latitude
+    )
+    if err then
+        log.error(err)
+
+        return bad_response(req, err.err, 500)
+    end
+
+    local forecast = build_forecast_object(raw_forecast, bucket_id, longitude, latitude)
+
+    -- trying to save into cache (tarantool)
+    local _, err = err_vshard_router:pcall(
+        router.call,
+        router,
+        bucket_id,
+        'write',
+        'weather_forecast_add',
+        {forecast}
+    )
+    if err then log.error(err) end
+
+    return ok_response(req, json.decode(forecast.raw_data))
+end
+
+-- validate config
+-- some basic validation
+exports.validate_config = function(cfg)
+    log.verbose('validate config')
+
+    -- HACK on startup - defaul value of provided config is nil
+    -- it's ok - we will handle it on apply_config
+    if cfg.weather_forecast_service_url ~= nil then
+        assert(type(cfg.weather_forecast_service_url) == 'string', 'weather_forecast_service_url must be a string')
+    end
+
+    return true
+end
+
+-- apply config
+exports.apply_config = function(conf)
+    log.verbose('apply config')
+
+    -- HACK if we get nil value - previous value saved
+    weather_forecast_service_url = conf.weather_forecast_service_url or weather_forecast_service_url
```

Comment:
I would like you to access clusterwide config in handler instead of using global var